.pre-commit-config.yaml

```diff
@@ -1,8 +1,18 @@
 # See https://pre-commit.com for more information
 # See https://pre-commit.com/hooks.html for more hooks
+
+files: |
```

Comment:
I guess this is the key part, but I'm shaky what this actually means ...
Does it somehow get "passed down" to configure the scope of the individual checking tasks,
or does it just configure what triggers the checks ?

If the latter, then I think we need more context-limitation.
I am finding that the presence of a .nox directory makes black take forever, so some excludes are needed.

Reply:
This took me a while to figure out from reading the `pre-commit` docs and sifting through their banked issues.

But essentially, yes, now `pre-commit` can filter files globally prior passing them to hooks. This is exactly the level of control that we need i.e., filter the changed files that we care about, rather than checking everything.

`black` is slightly different, but I've fixed the `.nox` issue by using `extend-exclude` rather than `exclude` in the `pyproject.toml` file... and `black` by default ignores `.nox` directories. Win.